sdk/core/Azure.Core/api/Azure.Core.net5.0.cs

```diff
@@ -256,6 +256,7 @@ public abstract partial class Response<T> : Azure.NullableResponse<T>
         protected Response() { }
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override bool HasValue { get { throw null; } }
+        public override T Value { get { throw null; } }
```

Comment:
The nullability is in direct conflict with the behavior of HasValue.

Reply:
This change keeps the nullability of Value consistent with the current API and the expected usage of the type. I think it would be much worse to advertise this property as nullable when it never should be and should never be going forward. The fact that a handful of APIs use this incorrectly should not be justification to advertise the bad edge case behavior, IMO.

Reply:
> I agree it would be worse to make this nullable, because that then invites the question "why do we even have NullableResponseOfT". However, I think it would be better to leave `Response<T>.HasValue` the way it is today, and fix the customer issue by subclassing `Response<T>` in the specific library where this issue occurs. I also like the suggestion from @annelo-msft to EBN HasValue if we are advertising `Response<T>` as the type to use when you always have a value (though now that I think about it I'm not sure if that's possible since HasValue is on the base type as well).
> 
> If we do decide to stick with the changes as is, I would ask that we add extensive comments explaining the history of why the properties work the way that they do, and also provide guidance to library authors on how to choose the correct Response type for their library.

We already have https://github.com/Azure/azure-sdk-for-net/issues/37171 to track this. Perhaps I am simplifying things here, now that we have  `NullableResponse<T>` using the correct `Response` type should be easier than ever. The confusion seems to be how to interpret the small number of incorrect usages that exist today.

> BTW is NullableResponse integrated with codegen/typespec/swagger?

Not yet, but [this issue ](https://github.com/Azure/autorest.csharp/issues/2159)seems relevant to tracking this

Reply:
> The confusion seems to be how to interpret the small number of incorrect usages that exist today.

The confusion is around what this means going forward for the various response types. We are changing the behavior of `Response<T>` to patch up these incorrect usages. But the change we are making makes the type functionally equivalent to `NullableResponse<T>`.